platforms/software/dependency-management/src/test/groovy/org/gradle/internal/component/local/model/TransformedComponentFileArtifactIdentifierTest.groovy

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.component.local.model
+
+
+import org.gradle.api.artifacts.component.ComponentIdentifier
+import spock.lang.Specification
+
+/**
+ * Tests {@link TransformedComponentFileArtifactIdentifier}
+ */
+class TransformedComponentFileArtifactIdentifierTest extends Specification {
+    def "has useful display name"() {
+        def componentId = newComponentId("foo")
+        def id = new TransformedComponentFileArtifactIdentifier(componentId, "current", "original")
+
+        expect:
+        id.getOriginalFileName() == "original"
+        id.getFileName() == "current"
+        id.getDisplayName() == "original -> current (foo)"
+    }
+
+    def "are equal when publish componentId and files are equal"() {
```

Comment:
💭 I think this test does too much.  It's named `are equal`, yet it's also testing several cases where the identifiers differ.  If this test failed, from looking at the name of the test in the test report, you wouldn't know which case had broken - finding equal things equal or finding non-equivalents not equal.  Also, the files aren't always equal in the comparisons being done here.

Reply:
Personally I think this is fine as a generic "if this test fails, something is wrong with the class's identity". All I want to test is that if they have the same contents they equals is true and they have the same hash code. I don't think this needs to be that much more complex. 

I see your point about the naming. I can split this up into equal and non equal tests if you want. LMK how you want this changed since IDC really how this ends up structured as long as it tests the same thing. 

What do you mean by `the files aren't always equal in the comparisons being done here.`? Isn't the point that the files are not equal when testing the non equal cases?

Reply:
> What do you mean by the files aren't always equal in the comparisons being done here.? Isn't the point that the files are not equal when testing the non equal cases?

Yes, that situation just contradicts the name.  I think either splitting into 2 methods or renaming it to something more inclusive would be fine.

Reply:
Renamed